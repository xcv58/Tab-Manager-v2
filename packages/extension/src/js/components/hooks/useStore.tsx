import React from 'react'
import Store from 'stores'

export const store = new Store()

export const StoreContext = React.createContext<Store>(store)

export const useStore = () => {
  return React.useContext(StoreContext)
}

export const useFontSize = () => {
  const { userStore } = useStore()
  return userStore.fontSize
}

export const useTabHeight = () => {
  const fontSize = useFontSize()
  return 3 * fontSize
}
