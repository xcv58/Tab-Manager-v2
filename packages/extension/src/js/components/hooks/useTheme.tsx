import { createContext, useContext } from 'react'

export const ThemeContext = createContext(false)

export const useTheme = () => useContext(ThemeContext)
