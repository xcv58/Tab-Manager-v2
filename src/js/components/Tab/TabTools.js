import React from 'react'
import { inject, observer } from 'mobx-react'
import CloseButton from 'components/Tab/CloseButton'
import DragHandle from 'components/Tab/DragHandle'
import TabMenu from 'components/Tab/TabMenu'

@inject('dragStore')
@observer
export default class TabTools extends React.Component {
  render () {
    const {
      faked,
      dragStore: { dragging },
      tab: { isHovered }
    } = this.props
    if (faked || dragging || !isHovered) {
      return null
    }
    return (
      <div
        style={{
          display: 'flex',
          justifySelf: 'flex-end'
          // position: 'absolute',
          // right: 0
        }}
      >
        <DragHandle {...this.props} />
        <TabMenu {...this.props} />
        <CloseButton {...this.props} />
      </div>
    )
  }
}
