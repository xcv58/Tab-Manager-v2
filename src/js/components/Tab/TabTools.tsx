import React from 'react'
import { observer } from 'mobx-react'
import DragHandle from 'components/Tab/DragHandle'
import TabMenu from 'components/Tab/TabMenu'
import { withStyles } from '@material-ui/core/styles'
import { useStore } from 'components/StoreContext'

const styles = () => ({
  root: {
    display: 'flex',
    justifySelf: 'flex-end'
  }
})

const TabTools = observer(props => {
  const { dragStore } = useStore()
  const {
    classes,
    faked,
    tab: { isHovered }
  } = props
  const { dragging } = dragStore
  if (faked || dragging || !isHovered) {
    return null
  }
  return (
    <div className={classes.root}>
      <DragHandle {...props} />
      <TabMenu {...props} />
    </div>
  )
})

export default withStyles(styles)(TabTools)
