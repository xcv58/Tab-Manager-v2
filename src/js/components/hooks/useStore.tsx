import React from 'react'
import Store from 'stores'

export const store = new Store()

export const StoreContext = React.createContext<Store>(store)

export const useStore = () => {
  return React.useContext(StoreContext)
}
