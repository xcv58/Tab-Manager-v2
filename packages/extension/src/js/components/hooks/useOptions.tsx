import { useStore } from 'components/hooks/useStore'

export const useOptions = () => {
  const { windowStore, shortcutStore, searchStore } = useStore()
  if (searchStore.isCommand) {
    const { shortcuts } = shortcutStore
    return shortcuts
      .map(([shortcut, command, name, hideFromCommand]) => {
        if (typeof name !== 'string' || hideFromCommand) {
          return null
        }
        return { name, shortcut, command }
      })
      .filter((x) => x)
      .sort((a, b) => a.name.localeCompare(b.name))
  }
  const { historyTabs } = searchStore
  console.log({ historyTabs: [...historyTabs] })
  return [...windowStore.tabs, ...historyTabs]
}
