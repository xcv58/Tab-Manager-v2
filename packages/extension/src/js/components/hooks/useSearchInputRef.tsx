import { useRef, useEffect } from 'react'
import { useStore } from 'components/hooks/useStore'

export const useSearchInputRef = () => {
  const searchInputRef = useRef<HTMLInputElement>(null)
  const { searchStore } = useStore()
  const { setSearchEl } = searchStore

  useEffect(() => setSearchEl(searchInputRef))
  return searchInputRef
}
