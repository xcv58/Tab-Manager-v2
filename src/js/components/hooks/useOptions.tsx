import { useStore } from 'components/hooks/useStore'

export const useOptions = () => {
  const { windowStore, shortcutStore, searchStore } = useStore()
  if (searchStore.isCommand) {
    const { shortcuts } = shortcutStore
    return shortcuts
      .map(([shortcut, command, name, hideFromCommand]) => {
        if (typeof name !== 'string' || hideFromCommand) {
          return
        }
        return { name, shortcut, command }
      })
      .filter((x) => x)
      .sort((a, b) => a.name.localeCompare(b.name))
  }
  return windowStore.tabs
}
