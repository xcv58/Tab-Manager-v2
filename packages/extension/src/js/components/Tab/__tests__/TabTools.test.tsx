import React from 'react'
import TabTools from 'components/Tab/TabTools'
import * as StoreContext from 'components/hooks/useStore'
import { cleanup, render, screen } from '@testing-library/react'

jest.mock('components/Tab/TabMenu', () => 'TabMenu')
const classes = { root: 'root' }
const props = {
  classes,
  faked: false,
  tab: { isHovered: true, removing: false },
}
const mockStore = {
  dragStore: { dragging: false },
}

describe('TabTools', () => {
  beforeEach(() => {
    jest.spyOn(StoreContext, 'useStore').mockImplementation(() => mockStore)
  })

  it('should render correct components', () => {
    const { container } = render(<TabTools {...props} />)
    expect(screen.getByRole('button')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  it('should render null', () => {
    let { container } = render(<TabTools {...props} faked />)
    expect(container.firstChild).toBeNull()

    cleanup()
    container = render(
      <TabTools {...props} tab={{ isHovered: false }} />
    ).container
    // expect(el.getElement()).toBe(null)
    expect(container.firstChild).toBeNull()

    jest
      .spyOn(StoreContext, 'useStore')
      .mockImplementation(() => ({ dragStore: { dragging: true } }))
    container = render(<TabTools {...props} />).container
    expect(container.firstChild).toBeNull()
  })
})
