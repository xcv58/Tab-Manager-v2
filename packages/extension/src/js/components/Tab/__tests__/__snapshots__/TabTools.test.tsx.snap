// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabTools should render correct components 1`] = `
<div>
  <div
    class="flex items-center"
  >
    <button
      class="inline-flex items-center justify-center w-8 h-8 m-2 rounded-full hover:shadow-xs focus:outline-none focus:ring cursor-move hover:bg-blue-200 active:bg-blue-300"
    >
      â˜°
    </button>
    <tabmenu
      classes="[object Object]"
      tab="[object Object]"
    />
  </div>
</div>
`;

exports[`TabTools should render null 1`] = `<div />`;
